Elephant factor

alpgarcia opened the pull request:
-----
* Resolves #2  
----
The Elephant Factor metric implementation is based on Pandas.
The architecture is divided in:
 - broomstick.data: access data layer, focused on ES by the time
   being.
 - broomstick.metrics: service layer providing metrics.
 - broomstick.core: common stuff needed by the data and service
   layers.

The notebooks folder contains a Jupyter notebook that uses
broomstick. Notebooks are the preferred way to use broomstick
library (up to now). We still need to define the visualization
part and decide whether broomstick is able to plot charts or
not. In this version it is, however there is a flag to not
plot anything if needed.

Tests have been implemented for all the available methods.

Commit:
Add Elephant Factor metric

The Elephant Factor metric implementation is based on Pandas.
The architecture is divided in:
 - broomstick.data: access data layer, focused on ES by the time
   being.
 - broomstick.metrics: service layer providing metrics.
 - broomstick.core: common stuff needed by the data and service
   layers.

The notebooks folder contains a Jupyter notebook that uses
broomstick. Notebooks are the preferred way to use broomstick
library (up to now). We still need to define the visualization
part and decide whether broomstick is able to plot charts or
not. In this version it is, however there is a flag to not
plot anything if needed.

Tests have been implemented for all the available methods.

Signed-off-by: Alberto Pérez García-Plaza <alpgarcia@bitergia.com>

Commit:
Add .vscode to the ignore list

Signed-off-by: Alberto Pérez García-Plaza <alpgarcia@bitergia.com>

Commit:
Add initial flake8 configuration

Ignore E402 and some common paths.

Signed-off-by: Alberto Pérez García-Plaza <alpgarcia@bitergia.com>

dicortazar commented:
Starting from scratch here!

Some improvements I see that might be valuable:

1- Create a minimum viable readme file that help to understand the goal of this. We can leave this as a further task if you prefer to.

2- Installation steps: I've needed to go through some steps when installing the virtual env. The following are my missing libraries that we may add to the installation process (or requirements file) at some point:

1- Create the virtual env instructions
2- Add missing libraries. In my case the following ones:
pip3 install pandas
pip3 install certifi
pip3 install cufflinks
pip3 install urllib3
pip3 install elasticsearch
pip3 install elasticsearch_dsl

Up to here and with this, the tests work for me :).

Then, in the notebook you indicate how to install broomstick, but I'm missing the setup.py file to do so. 


3- Typo

And just noticed in the notebook this says: print("Elphant Factor (excluding Unknown):", while I guess this should say Elephant.

Commit:
Add requirements file

Signed-off-by: Alberto Pérez García-Plaza <alpgarcia@bitergia.com>

Commit:
Update README file

Add more information about Broomstick architecture and goals.

Signed-off-by: Alberto Pérez García-Plaza <alpgarcia@bitergia.com>

alpgarcia commented:
> Starting from scratch here!
> 
> Some improvements I see that might be valuable:
> 
> 1- Create a minimum viable readme file that help to understand the goal of this. We can leave this as a further task if you prefer to.

Done.

> 
> 2- Installation steps: I've needed to go through some steps when installing the virtual env. The following are my missing libraries that we may add to the installation process (or requirements file) at some point:
> 
> 1- Create the virtual env instructions
> 2- Add missing libraries. In my case the following ones:
> pip3 install pandas
> pip3 install certifi
> pip3 install cufflinks
> pip3 install urllib3
> pip3 install elasticsearch
> pip3 install elasticsearch_dsl
> 

I've added the requirements file. By the time being there is no installation procedure for Broomstick, you need to make sure it is included in your python path.

> Up to here and with this, the tests work for me :).
> 
> Then, in the notebook you indicate how to install broomstick, but I'm missing the setup.py file to do so.
> 

There is only a `TODO` for this. The notebook is using `sys` to make sure Broomstick is in the path and can be imported.

> 3- Typo
> 
> And just noticed in the notebook this says: print("Elphant Factor (excluding Unknown):", while I guess this should say Elephant.

Fixed.

Please @dicortazar review my changes!

dicortazar commented:
+1, thanks.
